### PR TITLE
Set terminal output/input encodings to utf-8

### PR DESCRIPTION
**Pull request by ivanduplenskikh**
**Changes**: Set output and input encodings to utf-8 for Console
**WI**: 2079280

### CODE REVIEW

**Review by kirill-ivlev — APPROVED** [review]

**Review by DergachevE — APPROVED** [review]

File: src/Microsoft.VisualStudio.Services.Agent/Terminal.cs [hunk 1]
```diff
@@ -37,6 +39,25 @@ public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
             Console.CancelKeyPress += Console_CancelKeyPress;
+
+            var terminalEncoding = Encoding.UTF8;
```
Comment (merlynomsft) [inline src/Microsoft.VisualStudio.Services.Agent/Terminal.cs:43]: This should probably be behind a feature flag (off by default)

Comment (ivanduplenskikh) [reply src/Microsoft.VisualStudio.Services.Agent/Terminal.cs:43]: The main goal is to achieve compatibility with on-prem solutions as well. I think it should work by default.

File: src/Microsoft.VisualStudio.Services.Agent/Terminal.cs [hunk 2]
```diff
@@ -37,6 +39,25 @@ public override void Initialize(IHostContext hostContext)
...
+            var endEncodingName = AgentKnobs.AgentTerminalEncoding.GetValue(hostContext).AsString();
```
[the remark this answers is not in the document]
Comment (ivanduplenskikh) [reply src/Microsoft.VisualStudio.Services.Agent/Terminal.cs:44]: There are different UTF encodings such as UTF-8, UTF-16, UTF-32. Should we provide the mechanism to set these encodings as well?

**Review by kirill-ivlev — APPROVED** [review]

**Pull request merged**